### PR TITLE
Due date notification setting per board

### PR DESCRIPTION
**Pull request by juliushaertl**
* Resolves: #713
* Target version: master 

### Summary

- Implement a notification setting per board that allows to adjust if due date notifications should be sent
  - for all cards
  - only for assigned cards **default**
  - never

The user interface was slightly adjusted since having all those settings in the main menu feels to clutter a lot already, so [in contrast to Talk where the options are shown directly](https://github.com/nextcloud/deck/issues/713#issuecomment-598598598) this is now hidden in a submenu:

image

image

### TODO

- [x] Proper icons instead of radio buttons
- [x] Update API documentation for new config endpoints and extended board result
- [x] Check with @jancborchardt if we should make "all cards" the default for boards that are not shared yet
- [x] Cover assignement notification handling with tests

### Checklist

- [x] Code is properly formatted
- [x] Sign-off message is added to all commits
- [x] Tests (unit, integration, api and/or acceptance) are included
- [x] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: src/components/navigation/AppNavigationBoard.vue
Comment (nickvergessen) [inline src/components/navigation/AppNavigationBoard.vue:49]: This is :see_no_evil:

File: src/components/navigation/AppNavigationBoard.vue
Comment (nickvergessen) [inline src/components/navigation/AppNavigationBoard.vue:53]: btw trailing spaces inside the translation?

File: src/components/navigation/AppNavigationBoard.vue
Comment (nickvergessen) [inline src/components/navigation/AppNavigationBoard.vue:75]: not translated :scream:

File: src/components/navigation/AppNavigationBoard.vue
Comment (nickvergessen) [inline src/components/navigation/AppNavigationBoard.vue:81]: not translated :scream:

File: src/components/navigation/AppNavigationBoard.vue
Comment (nickvergessen) [inline src/components/navigation/AppNavigationBoard.vue:87]: not translated :scream: 

This will be weird to translate, make `No notifications` is better?

File: src/components/navigation/AppNavigationBoard.vue
Comment (juliushaertl) [reply src/components/navigation/AppNavigationBoard.vue:49]: :see_no_evil: Will clean up once the sub-menu is somewhat approved by @jancborchardt ;)

**Review by nickvergessen — CHANGES_REQUESTED** [review]
See comments

File: src/components/navigation/AppNavigationBoard.vue
Comment (juliushaertl) [reply src/components/navigation/AppNavigationBoard.vue]: Thanks, pushed a fix here to avoid conflicts, even though not very related, since it is a very minor change.

File: src/components/navigation/AppNavigationBoard.vue
Comment (skjnldsv) [inline src/components/navigation/AppNavigationBoard.vue:111]: Import components directly to gain some bytes?

Comment (juliushaertl) [reply src/components/navigation/AppNavigationBoard.vue:111]: When I checked last time we used so many of them that the combined bundle was actually smaller than including each of them, but i check that in irregular intervals.

**Review by ChristophWurst — APPROVED** [review]
Code looks clean

**Review by skjnldsv — APPROVED** [review]

**Comment by juliushaertl** [comment]
Quick update on the icons to align with talk:

image

**Comment by jancborchardt** [comment]
Very nice with the submenu! :) Is that a component already which could be used elsewhere?

Generally it’s super nice already, here’s some possibility for enhancement, your call if or when it gets done. :D 
- [ ] For proper feedback: A subline for the "Due date reminders" entry with the chosen setting (like you know from Android and iOS) because otherwise you have to go into the submenu to know the active setting. The icon could also reflect the chosen setting.
- [ ] Nice to have: A quick animation left/right when the submenu gets shown and when you go back.
- [ ] This is more generally about the ActionMenu component (cause this is also the case for Talk), but the hover/focus and active style needs to be adjusted to move from the blue bar & text opacity towards just changing the background. cc @ma12-co

**Comment by jancborchardt** [comment]
> Check with @jancborchardt if we should make "all cards" the default for boards that are not shared yet

Yeah, good question. Cause then would it switch to "Assigned cards" when you share it?

**Comment by juliushaertl** [comment]
> Yeah, good question. Cause then would it switch to "Assigned cards" when you share it? 

Yes, we could of course just hide the assigned setting if a board is not shared and just offer on/off then. Of course this will need some careful handling of adjusting the setting if all shares get removed from a board.

**Comment by jancborchardt** [comment]
> Yes, we could of course just hide the assigned setting if a board is not shared and just offer on/off then. Of course this will need some careful handling of adjusting the setting if all shares get removed from a board.

Right, then for a non-shared board it could just be a checkbox for "Due date notifications" without the need for a submenu.
Then the switch to a submenu with a "different" default setting wouldn’t be so strange, cause it’s more clear there’s a difference.

**Comment by juliushaertl** [comment]
@jancborchardt I'll leave the animation for a later point in time, other comments in scope of the deck app should be addressed

Peek 2020-11-06 13-05

File: src/components/navigation/AppNavigationBoard.vue
Comment (juliushaertl) [reply src/components/navigation/AppNavigationBoard.vue]: Refactored the submenu handling with the last commit

**Comment by stefan-niedermann** [comment]
@juliushaertl don't want to nag, but did you think about REST-API documentation (if this feature will also be available in the API)? 😇

**Comment by juliushaertl** [comment]
@stefan-niedermann I already thought about you :heart: 

https://github.com/nextcloud/deck/pull/2430/commits/b18ebe83b6c80e3db54e5eb07838d00b6d49c87c

**Pull request merged**
